pkg/service/backup_backend_s3.go

```diff
@@ -42,33 +45,39 @@ func (s *BackupBackendS3) writeState(state *model.BackupState) error {
 func (s *BackupBackendS3) FullBackupList() ([]model.BackupDetails, error) {
 	backupFolder := s.Path + "/" + model.FullBackupDirectory + "/"
 	s3prefix := "s3://" + s.bucket
+	lastRun := s.readState().LastRun
 	if s.backupPolicy.RemoveFiles != nil && *s.backupPolicy.RemoveFiles {
 		// when use RemoveFiles = true, backup data is located in backupFolder folder itself
 		files, _ := s.listFiles(backupFolder)
-		if len(files) > 0 {
-			return []model.BackupDetails{{
-				Key:          ptr.String(s3prefix + backupFolder),
-				LastModified: &s.readState().LastRun,
-				Size:         ptr.Int64(s.dirSize(backupFolder)),
-			}}, nil
+		if len(files) == 0 {
+			return []model.BackupDetails{}, nil
 		}
-		return []model.BackupDetails{}, nil
+		if s.fullBackupInProgress.Load() {
```

Comment:
Can be combined with the previous if.

Reply:
it's different scenarios, so I prefer to keep them separate